specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,764 @@
+        [propput] HRESULT SelectedCommandId([in] INT32 value);
+
+        /// Gets the selected CommandId.
+        [propget] HRESULT SelectedCommandId([out, retval] INT32* value);
+
+        /// Sets whether the `ContextMenuRequested` event is handled by host after
+        /// the event handler completes or if there is a deferral then after the deferral is completed.  
+        /// If Handled is set to TRUE then WebView2 will not display a context menu and will instead
+        /// use the SelectedCommandId property to indicate which, if any, context menu item to invoke.
+        /// If after the event handler or deferral completes Handled is set to FALSE then WebView2 
+        /// will display a context menu based on the contents of the MenuItems property. 
+        /// The default value is FALSE.
+        [propput] HRESULT Handled([in] BOOL value);
+
+        /// Gets whether the `ContextMenuRequested` event is handled by host.
+        [propget] HRESULT Handled([out, retval] BOOL* value);
+
+        /// Returns an `ICoreWebView2Deferral` object. Use this operation to
+        /// complete the event when the custom context menu is closed.
+        HRESULT GetDeferral([out, retval] ICoreWebView2Deferral** deferral);
+    }
+
+    /// Represents the information regarding the context menu. Includes the location of the request, 
+    /// the context selected and the appropriate data used for the actions of a context menu.
+    [uuid(b8611d99-eed6-4f3f-902c-a198502ad472), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuInfo : IUnknown
+    {
+        /// Gets the coordinates where the context menu request occured in relation to the upper left corner of the webview bounds.
+        [propget] HRESULT Location([out, retval] POINT* value);
+
+        /// Gets the kind of context that the user selected.
+        [propget] HRESULT ContextKind([out, retval] COREWEBVIEW2_CONTEXT_KIND* value);
+
+        /// Returns TRUE if the context menu was requested on the main frame and
+        /// FALSE if invoked on another frame.
+        [propget] HRESULT IsMainFrame([out, retval] BOOL* value);
+
+        /// Returns TRUE if the context menu is requested on a selection.
+        [propget] HRESULT HasSelection([out, retval] BOOL* value);
+
+        /// Returns TRUE if the context menu is requested on an editable component.
+        [propget] HRESULT IsEditable([out, retval] BOOL* value);
+
+        /// Returns TRUE if the context menu is requested on a component that contains a link.
+        [propget] HRESULT ContainsLink([out, retval] BOOL* value);
+
+        /// Gets the uri of the page.
+        [propget] HRESULT PageUri([out, retval] LPWSTR* value);
+
+        /// Gets the uri of the frame. Will match the PageUri if `get_IsMainFrame` is TRUE.
+        [propget] HRESULT FrameUri([out, retval] LPWSTR* value);
+
+        /// Gets the source uri of element (if context menu requested on a media element, null otherwise).
```

Comment:
I expect this is an absolute URI and so empty string is not a valid value, but it would make things easier and consistent if all of these conditional properties had Has{PropertyName} boolean properties. Similarly for LinkUri, LinkText.

Reply:
@Hosamtag we need to document what happens in the case of a video or picture or media element with multiple source element children.

We should add a HasSourceUri property

Reply:
It will display the active source.